provider/provider.go

```diff
@@ -189,6 +189,12 @@ func DatabricksProvider() *schema.Provider {
 				Default:     false,
 				Description: "Create ephemeral PAT tokens also for AZ CLI authenticated requests",
 			},
+			"azure_environment": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"DATABRICKS_AZURE_ENVIRONMENT", "ARM_ENVIRONMENT"}, "public"),
```

Comment:
read just from single standard `ARM_ENVIRONMENT`, no need to introduce new environment variable name.